src/toast/toaster.js

```diff
@@ -0,0 +1,161 @@
+// @flow
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import {PLACEMENT} from './constants';
+import {Root} from './styled-components';
+import Toast from './toast';
+import type {
+  ToasterPropsT,
+  ToasterStateT,
+  ToastPropsT,
+  ToastPropsAltT,
+} from './types';
+
+let toasterRef: React.Ref<typeof Toaster> = React.createRef();
+
+class ToasterComponent extends React.Component<ToasterPropsT, ToasterStateT> {
+  static defaultProps = {
+    placement: PLACEMENT.top,
+  };
+
+  state = {
+    toasts: [],
+  };
+
+  dismissHandlers = {};
+
+  toastId: number = 0;
+
+  onClose = (key: React.Key) => {
```

Comment:
can we use handler style that's introduced in https://github.com/uber-web/baseui/pull/366?

Reply:
Sure, something like `onClose = ({key: React.Key}) => {`?

Reply:
yes, that looks great. should we include the `event` too?